### DISCUSSION TITLE
Need help with dev stack install

### DISCUSSION THREAD

**Discussion by Impa10r** [RoboSats/robosats#1754] [Q&A]
**Describe the bug**
Having a hard time trying to set up a dev environment following the [setup.md](https://github.com/RoboSats/robosats/blob/main/setup.md) instructions. 

1. It says the http server will run on 8080, but it started on 8081. 
2. http://unsafe.testnet.robosats.com does not respond on any port. 
3. http://robosats.onion.moe redirects to http://wordle.ayra.ch:8081/wordle-err.html and displays error: 
![image](https://github.com/user-attachments/assets/164ee87d-5559-48cd-9b55-a0c2592f6406)

robosats.onion.moe:8081 opens Wordle Game

I did redirect robosats.onion.moe:8081/static/frontend/main.js to 127.0.0.1:8081/frontend/main.js, but still get Wordle Game at robosats.onion.moe:8081. Opening robosats.onion.moe:8081/static/frontend/main.js shows js code instead of Not Found, so the redirect definitely works. 

If I try to start my own back-end in docker, I get this:
```
~/src/robosats$ docker-compose up -d
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.backend.environment.DEVELOPMENT contains true, which is an invalid type, it should be a string, number, or a null
services.backend.build.args contains true, which is an invalid type, it should be a string, number, or a null
```

**Desktop (please complete the following information):**
 - OS: Ubuntu 22
 - Browser: Firefox

**Comment by Impa10r** [comment]
Update: replaced them with "true", now this:

```
~/src/robosats$ docker login -u impalor
Password:
Login Succeeded
~/src/robosats$ docker-compose up -d
Pulling telegram-watcher (backend-image:)...
ERROR: The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.

Continue with the new image? [yN]y
Pulling telegram-watcher (backend-image:)...
ERROR: pull access denied for backend-image, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

**Comment by Impa10r** [comment]
But I still need full instructions how to configure the dev backend manually. I cannot build docker every time I add a new line of code. I must be able to debug as I go.

**Comment by Impa10r** [comment]
Three weeks, no reaction...

**Comment by femelo** [comment]
> Update: replaced them with "true", now this:
> 
> ```
> ~/src/robosats$ docker login -u impalor
> Password:
> Login Succeeded
> ~/src/robosats$ docker-compose up -d
> Pulling telegram-watcher (backend-image:)...
> ERROR: The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.
> 
> Continue with the new image? [yN]y
> Pulling telegram-watcher (backend-image:)...
> ERROR: pull access denied for backend-image, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
> ```

@Impa10r , were you able to run with this?

**Comment by femelo** [comment]
@Impa10r, it worked for me stopping and removing previously built docker images `robosats-nodeapp-frontend-nginx`, `robosats-nodeapp-frontend-nginx` and `robosats-nodeapp-frontend-tor`, and composing only for the frontend:

```bash
cd robosats/frontend
npm install
npm run dev
# CTRL + C
cd ../nodeapp
docker compose up
```

**Comment by Impa10r** [comment]
> @Impa10r, it worked for me stopping and removing previously built docker images `robosats-nodeapp-frontend-nginx`, `robosats-nodeapp-frontend-nginx` and `robosats-nodeapp-frontend-tor`, and composing only for the frontend:
> 
> ```shell
> cd robosats/frontend
> npm install
> npm run dev
> # CTRL + C
> cd ../nodeapp
> docker compose up
> ```

I wanted to develop

**Comment by femelo** [comment]
> > @Impa10r, it worked for me stopping and removing previously built docker images `robosats-nodeapp-frontend-nginx`, `robosats-nodeapp-frontend-nginx` and `robosats-nodeapp-frontend-tor`, and composing only for the frontend:
> > ```shell
> > cd robosats/frontend
> > npm install
> > npm run dev
> > # CTRL + C
> > cd ../nodeapp
> > docker compose up
> > ```
> 
> I wanted to develop

This is a setup for frontend development (simple things). Incidentally the setup you used in your bug description is also for frontend development. If you want to run the backend you need to docker compose at the repo root.

**Comment by Impa10r** [comment]
> > > @Impa10r, it worked for me stopping and removing previously built docker images `robosats-nodeapp-frontend-nginx`, `robosats-nodeapp-frontend-nginx` and `robosats-nodeapp-frontend-tor`, and composing only for the frontend:
> > > ```shell
> > > cd robosats/frontend
> > > npm install
> > > npm run dev
> > > # CTRL + C
> > > cd ../nodeapp
> > > docker compose up
> > > ```
> > 
> > 
> > I wanted to develop
> 
> This is a setup for frontend development (simple things). Incidentally the setup you used in your bug description is also for frontend development. If you want to run the backend you need to docker compose at the repo root.

I want to debug and develop backend

**Comment by KoalaSat** [comment]
> > > > @Impa10r, it worked for me stopping and removing previously built docker images `robosats-nodeapp-frontend-nginx`, `robosats-nodeapp-frontend-nginx` and `robosats-nodeapp-frontend-tor`, and composing only for the frontend:
> > > > ```shell
> > > > cd robosats/frontend
> > > > npm install
> > > > npm run dev
> > > > # CTRL + C
> > > > cd ../nodeapp
> > > > docker compose up
> > > > ```
> > > 
> > > 
> > > I wanted to develop
> > 
> > 
> > This is a setup for frontend development (simple things). Incidentally the setup you used in your bug description is also for frontend development. If you want to run the backend you need to docker compose at the repo root.
> 
> I want to debug and develop backend

Then your best option is to use Docker
